### PR TITLE
✨Source Salesloft: migrate to low-code

### PR DESCRIPTION
**Pull request by pabloescoder**
Migrating Source Salesloft to YamlDeclarativeSource (Low-code) Airbyte CDK
Resolves https://github.com/airbytehq/airbyte-internal-issues/issues/7842

Migrated connector to low code
Used Inline Schema
Refactored Unit Tests

### CODE REVIEW

**Comment by pabloescoder** [comment]
The check for formatting errors pipeline is failing only on the unit_tests folder (which is excluded from the airbyte-ci path for formatting - [see here](https://github.com/airbytehq/airbyte/blob/master/pyproject.toml#L31C5-L31C17))
All the other files are properly formatted so we can ignore that.

File: airbyte-integrations/connectors/source-salesloft/source_salesloft/custom_authenticators.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+from dataclasses import dataclass
+from typing import Any, Mapping, Tuple
+
+import requests
+from airbyte_cdk.sources.declarative.auth import DeclarativeOauth2Authenticator
+
+
+@dataclass
+class Oauth2Authenticator(DeclarativeOauth2Authenticator):
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-salesloft/source_salesloft/custom_authenticators.py:13]: You need use `DeclarativeSingleUseRefreshTokenOauth2Authenticator` as refresh token expired frequently. After that I can run regression tests

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-salesloft/source_salesloft/custom_authenticators.py (hunk 1 above)
Comment (pabloescoder) [reply airbyte-integrations/connectors/source-salesloft/source_salesloft/custom_authenticators.py]: @lazebnyi I've changed the `OauthAuthenticator` to a `SingleUseOauthAuthenticator` that inherits the `DeclarativeSingleUseRefreshTokenOauth2Authenticator` class.
We're ready to run regression tests!

**Review by lazebnyi — APPROVED** [review]
LGTM

Regression tests passed locally as the customer has too many records - https://github.com/airbytehq/airbyte/actions/runs/9424360043/job/25964431616

**Pull request merged**